### ISSUE TITLE
Request for OpenSUSE Repo Support.

### ISSUE THREAD

**Issue by MooMew64** [sonic2kk/steamtinkerlaunch#726]
Is there any chance we could get STL with needed dependencies added to the OpenSUSE repos, or perhaps have a guide written for installation? I've been trying to use the install guide for the ProtonUpQT method but I get hung up on the part where you have to install -yad, as yad is unfortunately not in OpenSUSE's repos. ~~Compiling yad manually also does not seem to work as all that happens when I try is compiling fails and adds a .deps folder that changes its folder permissions to 'root' and becomes irremovable from my system unless I use sudo rm -rf. I'm going to have to use the appimage, but I'm concerned for longterm support with that given it isn't maintained.~~ EDIT: I was able to get yad to compile, needed to use su over sudo. doh

That said, if possible, is there a way to submit this to the OpenSUSE repos? It would help out a lotta folks like me who are running OpenSUSE for OTB RDNA3 support atm. I understand if it isn't possible, though.

Just also wanna add how much I appreciate this project as a Linux fan and a Sonic fan; This is a loads better route to modding than some of my earliest Steam Deck experiments, and is helping a whole new community of people get the most out of their funni fast hedgehog experience on Linux.

**Comment by sonic2kk** [comment]
I'm glad you like SteamTinkerLaunch and thank you for the kind words!

> That said, if possible, is there a way to submit this to the OpenSUSE repos? It would help out a lotta folks like me who are running OpenSUSE for OTB RDNA3 support atm. I understand if it isn't possible, though.

I haven't used OpenSUSE before (though I'd like to try it out at some point) but I would love to see OpenSUSE repo support! Having said that, I don't maintain distro packages for SteamTinkerLaunch. They're all maintained by kind folks that volunteer their time. I try to keep up with a couple that I am able to such as the AUR package (which I believe I'm a co-maintainet of now) and in the past I have reached out to the Pacstall author.

I have absolutely nothing against there being an OpenSUSE package, in fact I would love to see more packages! Specifically because I know Yad can be a pain-point. However since I don't use OpenSUSE and because I don't maintain any SteamTinkerLaunch packages officially, this isn't something that *I* can give support for. However, to any dedicated OpenSUSE users out there, please feel free :-)

I'm happy to leave this open, and when I have some time later I'd also be happy to add some specific steps to the Installation wiki with a note that Yad has to be installed manually.

Thanks!

**Comment by sonic2kk** [comment]
I just wanted to add as well that if someone (or yourself, please feel free if you know how and would like to!) wants to create a package but needs any involvement from me for some reason that I can try to help out. It's just that I don't use OpenSUSE so I don't know the process of creating/maintaining packages, and since I don't use it it's probably not a great idea for me to be a maintainer 😅 

Basically I'd love to see it and will try to help in any way needed to make this happen (if necessary) :-)

**Comment by sonic2kk** [comment]
I did some quick searching and found that there is a maintainer for a SteamTinkerLaunch OpenSUSE package. They also ran into the Yad issue and actually went to the effort of [packaging a version of Yad as well](https://build.opensuse.org/package/show/home:VortexAcherontic/yad). However, the version they package is slightly behind, still on [v11.11](https://build.opensuse.org/package/show/home:VortexAcherontic/steamtinkerlaunch). The Yad version is slightly behind as well, but I believe this is around the same version we use on Steam Deck, and no newer Yad functionality is taken advantage of really, so it is not really a big deal. Just some background :smile: 

I don't know very much about OpenSUSE but I would guess this OpenSUSE Builds platform is a bit like Fedora Copr? I don't see any mention of this build platform on [repology](https://repology.org/repositories/statistics). In other words this is kind of an extra-unofficial platform, and you are requesting a more mainline package be made available for SteamTinkerLaunch. Maybe I'm wrong though :sweat_smile: 

There are a few things you could do here:
- Ask the maintainer of the package I linked to update SteamTinkerLaunch to v12
- Request a SteamTinkerLaunch package somewhere more appropriate where a maintainer would be able to pick it up and put it on a repo (e.g. a repo that could be listed on repology?)
- Request a more up-to-date Yad package be included/provided in the repos for OpenSUSE going forward, that way ProtonUp-Qt could become a viable installation method (though personally I still like seeing packages available as well :-))

In the meantime though, I would like to add the steps for OpenSUSE to the wiki. But given my unfamiliarity with the distro I would like if you could outline the basic steps required for installation - I guess the only thing required is manually installing Yad?

I would also like to know if it's worth mentioning that repository I linked to on the wiki, I am not sure how useful that is to OpenSUSE users or if it is easily available via a package manager, but if it's worth directing users to I would be happy to do so.

Once the steps are added to the wiki I think I will close this issue, as aside from what I've outlined here (request an update to the OpenSUSE Build package, request a STL/Yad package be provided further upstream, and provide some basic installation guidance) I don't think much else can be done on my side.

Thanks!

**Comment by sonic2kk** [comment]
@SolarNyan bump :-)

There is another community package that seems to be up-to-date on v12.0: https://build.opensuse.org/package/show/home%3Asyntobu%3AGMR/steamtinkerlaunch

Again, these are community builds and not necessarily endorsements or anything. Not sure how to proceed and I am not sure something like STL would ever get into "official" repos for a distro like OpenSUSE.

I added a note about these packages as well as a note that Yad will have to be installed/managed separately onto the [installation wiki](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Installation#opensuse).

Let me know if this resolves the issue and if not, what further work has to be done :-) Otherwise we can close this issue I think.

**Comment by MooMew64** [comment]
I'm so sorry I haven't gotten back to this yet; Kept meaning to but I kept putting it off. My bad. >.<

Anyways, I appreciate all the time you've taken to look into this! It'll go a long way into helping other OpenSUSE users.

As for how I ended up installing it, I believe most of the dependecies are in OpensuseOSS (their main repo), but the names can vary. Looking through I would probably recommend anyone doing this from scratch to:

-Open YaST
 -Download the following if you don't already have them: gawk, bash, git, grep, unzip, wget, xdotool, xprop, xrandr, xxd, xwininfo
-yad has to be compiled from source, unfortunately. Follow the steps on Github page for yad 
-Install autoconf, automake, intltool, and GTK3 with all it's dev versions. GTK4 might also work, or you can grab all the GTKs OpenSUSE offers (2, 3, and 4) just to be safe (that's what I did.)
-When following yad's compiling guide, run _this_ before you compile in your yad folder:

```
su

```

then...

```
autoreconf -ivf && intltoolize

```

```

./configure && make && make install

```

OpenSUSE needs su, otherwise, a dependency folder will be inaccessible and compiling will fail. After that, I simply installed STL with ProtonUp-Qt's Flatpak and it works flawlessly! Only negative ATM is Sonic Frontiers is kinda borked on RDNA3 right now (I'm having REALLY BAD rendering glitches on my 7900XTX and current Mesa), but that's not a STL issue lol. 

This is how I was able to install everything. Hopefully this will help other SUSE users out! It's a fantastic distro, just suffers from a weird package manager and unfortunate codec situations (although the recent inclusion of open source versions of h.264 is a HUGE step towards ending that for good).

**Comment by sonic2kk** [comment]
Thanks for the great detailed steps! They all sound reasonable enough to me.

I [added them to the wiki](https://github.com/sonic2kk/steamtinkerlaunch/wiki/Installation#opensuse) crediting them to you and linking to this issue. The page is a bit long but there is a table of contents to hopefully help users find the steps that they're looking for :sweat_smile: 

> After that, I simply installed STL with ProtonUp-Qt's Flatpak and it works flawlessly!

Nice! I also noted that installing with ProtonUp-Qt afterwards will work on the instructions, in case the manual Yad installation happens to throw any users off. Me and DavidoTek put a lot of effort into the SteamTinkerLaunch support (back before I was even an STL maintainer!) so it's nice to see it get good use :-)

> Only negative ATM is Sonic Frontiers is kinda borked on RDNA3 right now

I actually stumbled a DXVK issue for this the other day when browsing its issue tracker (something something "Sonic the Hedgehog, snooping as usual"). Here's hoping it gets fixed soon :-)

I'll close this issue now as I'm not sure if there's anything more that can be done from my side for OpenSUSE repo support.

Thanks!

**Comment by biggestsonicfan** [comment]
SUSE user here, I needed the `intltool` and `gtk3-devel` packages, but I did not make YAD as root. Instead, `./configure` and `make` were run as local user, and `sudo make install` was used to install YAD. This I believe is the common and preferred method, it's very unusual if you need to configure and make as root, if you do that usually means there's a dependency installation issue. I would also recommend using `zypper in` for package installation over YAST, as that will most likely speed up the install process and add fewer steps.
